api/main.go

```diff
@@ -582,7 +585,40 @@ func (serv *MetadataServer) ListProviders(in *pb.Empty, stream pb.Api_ListProvid
 
 func (serv *MetadataServer) CreateProvider(ctx context.Context, provider *pb.Provider) (*pb.Empty, error) {
 	serv.Logger.Infow("Creating Provider", "name", provider.Name)
-	return serv.meta.CreateProvider(ctx, provider)
+	_, err := serv.meta.CreateProvider(ctx, provider)
+	if err != nil {
+		serv.Logger.Errorw("Failed to create provider", "error", err)
+		return nil, err
+	}
+	var status *pb.ResourceStatus
```

Comment:
we have a setResourceStatus function that I think we should leverage.

Reply:
What's your thinking on where to add this health check. We have CreateProvider in api/main.go, metadata/client.go and metadata.go?